File: frontend/lib/data/models/trail.dart
class Trail {
  final String id;
  final String name;
  final String category;
  final String description;
  final List<String> stops;
  final List<String> guideIds;

  Trail({
    required this.id,
    required this.name,
    required this.category,
    required this.description,
    required this.stops,
    required this.guideIds,
  });

  factory Trail.fromJson(Map<String, dynamic> json) {
    return Trail(
      id: json['id'],
      name: json['name'],
      category: json['category'],
      description: json['description'],
      stops: List<String>.from(json['stops'] ?? []),
      guideIds: List<String>.from(json['guideIds'] ?? []),
    );
  }

  Map<String, dynamic> toJson() => {
    'id': id,
    'name': name,
    'category': category,
    'description': description,
    'stops': stops,
    'guideIds': guideIds,
  };
}

File: wander-oromia-backend/models/Trail.js
// models/Trail.js
const mongoose = require("mongoose");

const trailSchema = new mongoose.Schema(
  {
    title: { type: String, required: true },
    category: {
      type: String,
      enum: ["Festivals", "Food & Cuisine", "Nature & Eco", "Wildlife"],
      required: true,
    },
    description: { type: String },
    imageUrl: { type: String }, // relative or full URL
    stops: [{ type: mongoose.Schema.Types.ObjectId, ref: "Stop" }],
  },
  { timestamps: true }
);

module.exports = mongoose.model("Trail", trailSchema);
